### PR TITLE
feat(snackbar): add swipe gesture to snackbar

### PR DESCRIPTION
**Pull request by soykje**
**TASK**: #1893 

### Description, Motivation and Context
We want to support swipe gesture on our component. For now, we only want the Snackbar to be closed on left/right gestures.

### Types of changes
- [x] ✨ New feature (non-breaking change which adds functionality)
- [x] 🧪 Test
- [x] 🧠 Refactor
- [x] 💄 Styles

### CODE REVIEW

File: packages/components/snackbar/src/useSwipe.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    swipeElement.addEventListener('pointerdown', onSwipeStart)
+    swipeElement.addEventListener('pointermove', onSwipeMove)
+    swipeElement.addEventListener('pointerup', onSwipeEnd)
+
+    return () => {
+      swipeElement.removeEventListener('pointerdown', onSwipeStart)
+      swipeElement.removeEventListener('pointermove', onSwipeMove)
+      swipeElement.removeEventListener('pointerup', onSwipeEnd)
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (acd02) [inline packages/components/snackbar/src/useSwipe.ts:74]: Why not include all the exhaustive dependencies?

File: packages/components/snackbar/src/useSwipe.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+interface UseSwipeReturn {
+  state?: 'start' | 'move' | 'end'
+  direction?: 'up' | 'down' | 'right' | 'left'
+}
+
+export const useSwipe = <T extends HTMLElement>({
+  swipeRef,
+  threshold = 10,
+}: UseSwipeArgs<T>): UseSwipeReturn => {
+  const [state, setState] = useState<UseSwipeReturn['state'] | undefined>()
```
Comment (acd02) [inline packages/components/snackbar/src/useSwipe.ts:17]: 💅
```suggestion
  const [state, setState] = useState<UseSwipeReturn['state']>()
```

File: packages/components/snackbar/src/useSwipe.ts [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+  const [direction, setDirection] = useState<UseSwipeReturn['direction'] | undefined>()
```
Comment (acd02) [inline packages/components/snackbar/src/useSwipe.ts:18]: 💅
```suggestion
  const [direction, setDirection] = useState<UseSwipeReturn['direction']>()
```

**Comment by soykje** [comment]
@acd02 This is still a draft, so please don't review! :stuck_out_tongue_closed_eyes:

**Review by Powerplex — APPROVED** [review]

**Review by andresin87 — APPROVED** [review]

**Review by turolopezsanabria — APPROVED** [review]
thnx!
The swipe gesture works like a charm on desktop and mobile
✅ OK design review for this PR

**Review by andresin87 — APPROVED** [review]

**Review by turolopezsanabria — APPROVED** [review]
Brilliant, I've just checked,
✅ Design Review done

**Pull request merged**